src/elements/content-uploader/ContentUploader.js

```diff
@@ -628,21 +628,24 @@ class ContentUploader extends Component<Props, State> {
      */
     removeFileFromUploadQueue = (item: UploadItem) => {
         const { onCancel, useUploadsManager } = this.props;
+        const { items, view } = this.state;
         // Clear any error errorCode in footer
         this.setState({ errorCode: '' });
 
         const { api } = item;
         api.cancel();
 
-        const { items } = this.state;
         items.splice(items.indexOf(item), 1);
 
         // Minimize uploads manager if there are no more items
         const callback = useUploadsManager && !items.length ? this.minimizeUploadsManager : noop;
 
         onCancel([item]);
         this.updateViewAndCollection(items, callback);
-        this.upload();
+
+        if (view === VIEW_UPLOAD_IN_PROGRESS) {
+            this.upload();
+        }
```

Comment:
would it make sense to add this guard check for other calls to `this.upload()`? Specifically, `handleUploadSuccess` and `handleUploadError`?